src/main/java/com/google/devtools/build/lib/actions/FileArtifactValue.java

```diff
@@ -539,16 +542,27 @@ public static class RemoteFileArtifactValue extends FileArtifactValue {
     private final long size;
     private final int locationIndex;
     private final String actionId;
-
-    public RemoteFileArtifactValue(byte[] digest, long size, int locationIndex, String actionId) {
+    @Nullable private final PathFragment symlinkTargetExecPath;
```

Comment:
That may not be that useful internally. Both this and the actionid, please check the impact on memory usage due to that.

Reply:
Introduced a separate `SymlinkToRemoteFileArtifact` subclass to avoid increasing the size. For `TreeArtifactValue`, we got lucky and the new field fits within the preexisting padding, so no subclass is required.